components/insight/SRC/org/openmicroscopy/shoola/env/data/OMEROGateway.java

```diff
@@ -1814,13 +1816,18 @@ client createSession(String userName, String password, String hostName,
 	{
 		this.encrypted = encrypted;
 		client secureClient = null;
+		
 		try {
 		    // client must be cleaned up by caller.
 			if (port > 0) secureClient = new client(hostName, port);
 			else secureClient = new client(hostName);
 			secureClient.setAgent(agentName);
-			ServiceFactoryPrx entryEncrypted
-			    = secureClient.createSession(userName, password);
+			ServiceFactoryPrx entryEncrypted;
+			if (userName.equals(password)) {
```

Comment:
If guest isn't working, I really think this block or even the simpler `createSession(username, password)` is fine considering the implementation of `joinSession`:

https://github.com/openmicroscopy/openmicroscopy/blob/develop/components/blitz/src/omero/client.java#L635